src/test/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndDeleteCommandResolverTest.java

```diff
@@ -205,5 +208,27 @@ public void idFilterWithProjection() throws Exception {
                         });
               });
     }
+
+    @Test
+    public void idFilterWithProjectionSimilarity() throws Exception {
+      String json =
+          """
+                  {
+                    "findOneAndDelete": {
+                      "filter" : {"_id" : "id"},
+                      "projection" : { "$similarity" : 1 }
```

Comment:
Something worth considering: it may be worthwhile to require using `$similarity: { $meta: 1 }` or something similar, instead of `$similarity: 1`. Because, otherwise, `$similarity: 1` looks like an inclusive projection, which makes it difficult to tell JSON API "get me the document with all fields, plus `$similarity`".

Mongoose [has logic to avoid treating objects in projections as a clause that defines whether the projection as inclusive or exclusive](https://github.com/Automattic/mongoose/blob/7287420d94e7917049ad78b6ce0e03d9459473ad/lib/helpers/projection/isDefiningProjection.js#L12-L16). Even if JSON API treats `$similarity` as a special key that doesn't define the projection as inclusive or exclusive, we would also have to add support for that to Mongoose. Which is why I think the `$meta` approach is worth considering.

Reply:
@vkarpov15 The design was to ask for required column when similarity score is needed. Should we add it as a command options also, so it can return along with full document?